src/ADNLPProblems/minsurfo.jl

```diff
@@ -0,0 +1,114 @@
+#  Find the surface with minimal area, given boundary conditions, 
+#  and above an obstacle.
+
+#  This is problem 17 in the COPS (Version 3) collection of 
+#  E. Dolan and J. More'
+#  see "Benchmarking Optimization Software with COPS"
+#  Argonne National Labs Technical Report ANL/MCS-246 (2004)
+#  classification OBR2-AN-V-V
+
+using Plots
+using ADNLPModels, NLPModels, NLPModelsIpopt, DataFrames, LinearAlgebra, Distances, SolverCore, PyPlot
+
+function minsurfo(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
+    
+    # domain definition
+    xmin = T(0.)
+    xmax = T(1.)
+    ymin = T(0.)
+    ymax = T(1.)
+
+    # Definition of the mesh
+    nx = 20 # number of points according to the direction x
+    ny = 20 # number of points according to the direction y
```

Comment:
Is there a reason to choose 20? I suggest we do: `nx = Int(round(sqrt(n)))` and `ny = nx`